### PR TITLE
machine credential fixes

### PR DESCRIPTION
**Pull request by poettering**
This is a follow-up for e8ac916ec35566cc1daab767ba3fe9449ad25eca

It fixes some logging issues, memory corruption on error path issues and more.

Material for v255

### CODE REVIEW

File: src/shared/machine-credential.c [hunk 1]
```diff
@@ -40,7 +40,7 @@ int machine_credential_set(MachineCredential **credentials, size_t *n_credential
 
         r = extract_first_word(&p, &word, ":", EXTRACT_DONT_COALESCE_SEPARATORS);
         if (r == -ENOMEM)
-                return r;
+                return log_oom();
```
Comment (YHNdnzj) [inline src/shared/machine-credential.c]: We don't usually specialize `ENOMEM`?

File: src/shared/machine-credential.c [hunk 2]
```diff
@@ -86,7 +86,7 @@ int machine_credential_load(MachineCredential **credentials, size_t *n_credentia
 
         r = extract_first_word(&p, &word, ":", EXTRACT_DONT_COALESCE_SEPARATORS);
         if (r == -ENOMEM)
-                return -ENOMEM;
+                return log_oom();
```
Comment (YHNdnzj) [inline src/shared/machine-credential.c]: also here. Let's just log everything in `log_error_errno` below.

File: src/shared/machine-credential.c [hunk 3]
```diff
@@ -30,13 +30,11 @@ void machine_credential_free_all(MachineCredential *creds, size_t n) {
 
 int machine_credential_set(MachineCredential **credentials, size_t *n_credentials, const char *cred_string) {
```
Comment (YHNdnzj) [inline src/shared/machine-credential.c:31]: We usually introduce `MachineCredentials` or `-Context` for this. But can be done later.

**Review by bluca — APPROVED** [review]

File: src/shared/machine-credential.c [hunk 4]
```diff
@@ -30,13 +30,11 @@ void machine_credential_free_all(MachineCredential *creds, size_t n) {
...
         _cleanup_free_ char *word = NULL, *data = NULL;
-        MachineCredential *creds = *ASSERT_PTR(credentials);
         ssize_t l;
-        size_t n_creds = *ASSERT_PTR(n_credentials);
         int r;
         const char *p = ASSERT_PTR(cred_string);
 
-        assert(creds || n_creds == 0);
+        assert(credentials || n_credentials == 0);
```
Comment (YHNdnzj) [inline src/shared/machine-credential.c]: n_credentials needs to be dereferenced

Comment (poettering) [reply src/shared/machine-credential.c]: ouch

**Comment by poettering** [comment]
updated, ptal

**Review by YHNdnzj — APPROVED** [review]

**Review by yuwata — APPROVED** [review]

**Pull request merged**